### PR TITLE
chore: Use `_FILENAME` to download archive

### PR DESCRIPTION
**Pull request by Diadlo**
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/4980)

### CODE REVIEW

**Comment by anthonybilinski** [comment]
:lgtm_strong:

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4980#-:-L6cJxFiILo49TFZ8Ohn:b-jbsf8z)*

**Comment by nurupo** [comment]
You probably need to update

```
EXIF_HASH="16cdaeb62eb3e6dfab2435f7d7bccd2f37438d21c5218ec4e58efa9157d4d41a"
```

**Comment by Diadlo** [comment]
@nurupo Thanks. Already copied, but forgot to paste :)

File: windows/cross-compile/build.sh [hunk 1]
```diff
@@ -742,19 +742,20 @@ fi
...
 if [ ! -f "$EXIF_PREFIX_DIR/done" ]
 then
   rm -rf "$EXIF_PREFIX_DIR"
   mkdir -p "$EXIF_PREFIX_DIR"
 
-  wget https://sourceforge.net/projects/libexif/files/libexif/0.6.21/libexif-$EXIF_VERSION.tar.bz2
+  wget https://github.com/libexif/libexif/archive/libexif-${EXIF_VERSION}-release.tar.gz
   check_sha256 "$EXIF_HASH" "libexif-$EXIF_VERSION.tar.bz2"
   bsdtar --no-same-owner --no-same-permissions -xf libexif*.tar.bz2
   rm libexif*.tar.bz2
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Update archive name

File: windows/cross-compile/build.sh [hunk 2]
```diff
@@ -742,19 +742,20 @@ fi
...
 EXIF_HASH="16cdaeb62eb3e6dfab2435f7d7bccd2f37438d21c5218ec4e58efa9157d4d41a"
 if [ ! -f "$EXIF_PREFIX_DIR/done" ]
 then
   rm -rf "$EXIF_PREFIX_DIR"
   mkdir -p "$EXIF_PREFIX_DIR"
 
-  wget https://sourceforge.net/projects/libexif/files/libexif/0.6.21/libexif-$EXIF_VERSION.tar.bz2
+  wget https://github.com/libexif/libexif/archive/libexif-${EXIF_VERSION}-release.tar.gz
   check_sha256 "$EXIF_HASH" "libexif-$EXIF_VERSION.tar.bz2"
   bsdtar --no-same-owner --no-same-permissions -xf libexif*.tar.bz2
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Update archive name

File: windows/cross-compile/build.sh [hunk 3]
```diff
@@ -742,19 +742,20 @@ fi
...
+EXIF_VERSION=0_6_21
 EXIF_HASH="16cdaeb62eb3e6dfab2435f7d7bccd2f37438d21c5218ec4e58efa9157d4d41a"
 if [ ! -f "$EXIF_PREFIX_DIR/done" ]
 then
   rm -rf "$EXIF_PREFIX_DIR"
   mkdir -p "$EXIF_PREFIX_DIR"
 
-  wget https://sourceforge.net/projects/libexif/files/libexif/0.6.21/libexif-$EXIF_VERSION.tar.bz2
+  wget https://github.com/libexif/libexif/archive/libexif-${EXIF_VERSION}-release.tar.gz
   check_sha256 "$EXIF_HASH" "libexif-$EXIF_VERSION.tar.bz2"
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Update archive name.

File: windows/cross-compile/build.sh [hunk 4]
```diff
@@ -742,19 +742,20 @@ fi
 # Exif
 
 EXIF_PREFIX_DIR="$DEP_DIR/libexif"
-EXIF_VERSION=0.6.21
+EXIF_VERSION=0_6_21
 EXIF_HASH="16cdaeb62eb3e6dfab2435f7d7bccd2f37438d21c5218ec4e58efa9157d4d41a"
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Update hash.

Comment (Diadlo) [reply windows/cross-compile/build.sh]: Done

File: windows/cross-compile/build.sh (hunk 3 above)
Comment (Diadlo) [reply windows/cross-compile/build.sh]: Done

File: windows/cross-compile/build.sh (hunk 2 above)
Comment (Diadlo) [reply windows/cross-compile/build.sh]: Glob is still valid

File: windows/cross-compile/build.sh (hunk 1 above)
Comment (Diadlo) [reply windows/cross-compile/build.sh]: Glob is still valid

**Comment by nurupo** [comment]
If you are going to add `EXIF_FILENAME` then add it for all the deps: Qt, openssl, etc., so that everything is consistent.

**Comment by nurupo** [comment]
Please don't replace the `bsdtar --no-same-owner --no-same-permissions` command with a plain `tar`, it breaks everything, there is a reason why we changed `tar` to `bsdtar` and also added those `--no-same-owner --no-same-permissions`.

**Comment by nurupo** [comment]
Specifically, you are re-introducing bugs fixed by https://github.com/qTox/qTox/pull/4892 and https://github.com/qTox/qTox/pull/4571.

**Comment by Diadlo** [comment]
Argh... Hate it

**Comment by nurupo** [comment]
You just had to update the link, filename and hash, no need to make a separate `EXIF_FILENAME` variable if you don't want to add `_FILENAME`  for the rest of dependencies that the script wgets so that the script is consistent, and no need to change the tar command. The `_FILENAME` change would be nice to have though.

**Comment by Diadlo** [comment]
@nurupo Can you update your [Makefile](https://github.com/nurupo/mingw-w64-debug-scripts/blob/master/Makefile)?

**Comment by nurupo** [comment]
Update in what way?

**Comment by nurupo** [comment]
The URLs and hashes in the Makefile are correct, it's just SF redirecting to the "website under maintenance" page when trying to wget gdb, much like it has been happening with libexif.

**Comment by Diadlo** [comment]
@nurupo It's exactly what I meant, can you change Makefile to download not from SF?

**Comment by sudden6** [comment]
@Diadlo I don't think mingw is hosted anywhere else.

Also SF seems to be back up and running https://twitter.com/sfnet_ops/status/970143157307559936

**Comment by nurupo** [comment]
@Diadlo any suggestion as to where should I download from instead? AFAIK mingw hosts things primarily (only?) on SF, so I don't know where else it can be downloaded from.

**Comment by Diadlo** [comment]
Since FS fixed, removed all libexif commits

**Comment by sudden6** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4980)*

**Comment by nurupo** [comment]
Is this WIP? Because this doesn't look right.

File: windows/cross-compile/build.sh [hunk 5]
```diff
@@ -220,14 +220,15 @@ strip_all()
...
   rm -rf "$OPENSSL_PREFIX_DIR"
   mkdir -p "$OPENSSL_PREFIX_DIR"
 
-  wget https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz
-  check_sha256 "$OPENSSL_HASH" "openssl-$OPENSSL_VERSION.tar.gz"
-  bsdtar --no-same-owner --no-same-permissions -xf openssl*.tar.gz
+  wget "https://www.openssl.org/source/$OPENSSL_FILENAME"
+  check_sha256 "$OPENSSL_HASH" "$OPENSSL_FILENAME"
+  bsdtar --no-same-owner --no-same-permissions -xf "$OPENSSL_FILENAME"
   rm openssl*.tar.gz
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

File: windows/cross-compile/build.sh [hunk 6]
```diff
@@ -262,12 +263,13 @@ QT_MINOR=9
...
 QT_HASH="57acd8f03f830c2d7dc29fbe28aaa96781b2b9bdddce94196e6761a0f88c6046"
+QT_FILENAME="qt-everywhere-opensource-src-$QT_VERSION.tar.xz"
 if [ ! -f "$QT_PREFIX_DIR/done" ]
 then
   rm -rf "$QT_PREFIX_DIR"
   mkdir -p "$QT_PREFIX_DIR"
 
-  wget https://download.qt.io/official_releases/qt/$QT_MAJOR.$QT_MINOR/$QT_VERSION/single/qt-everywhere-opensource-src-$QT_VERSION.tar.xz
+  wget "https://download.qt.io/official_releases/qt/$QT_MAJOR.$QT_MINOR/$QT_VERSION/single/$QT_FILENAME"
   check_sha256 "$QT_HASH" "qt-everywhere-opensource-src-$QT_VERSION.tar.xz"
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

File: windows/cross-compile/build.sh [hunk 7]
```diff
@@ -262,12 +263,13 @@ QT_MINOR=9
...
   bsdtar --no-same-owner --no-same-permissions -xf qt*.tar.xz
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

File: windows/cross-compile/build.sh [hunk 8]
```diff
@@ -262,12 +263,13 @@ QT_MINOR=9
...
   rm qt*.tar.xz
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

File: windows/cross-compile/build.sh [hunk 9]
```diff
@@ -439,15 +442,16 @@ fi
...
 FFMPEG_HASH="1131d37890ed3dcbc3970452b200a56ceb36b73eaa51d1c23c770c90f928537f"
+FFMPEG_FILENAME="ffmpeg-$FFMPEG_VERSION.tar.xz"
 if [ ! -f "$FFMPEG_PREFIX_DIR/done" ]
 then
   rm -rf "$FFMPEG_PREFIX_DIR"
   mkdir -p "$FFMPEG_PREFIX_DIR"
 
-  wget https://www.ffmpeg.org/releases/ffmpeg-$FFMPEG_VERSION.tar.xz
+  wget "https://www.ffmpeg.org/releases/$FFMPEG_FILENAME"
   check_sha256 "$FFMPEG_HASH" "ffmpeg-$FFMPEG_VERSION.tar.xz"
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

File: windows/cross-compile/build.sh [hunk 10]
```diff
@@ -710,14 +714,15 @@ fi
...
   rm -rf "$QRENCODE_PREFIX_DIR"
   mkdir -p "$QRENCODE_PREFIX_DIR"
 
-  wget https://fukuchi.org/works/qrencode/qrencode-$QRENCODE_VERSION.tar.bz2
-  check_sha256 "$QRENCODE_HASH" "qrencode-$QRENCODE_VERSION.tar.bz2"
-  bsdtar --no-same-owner --no-same-permissions -xf qrencode*.tar.bz2
+  wget https://fukuchi.org/works/qrencode/$QRENCODE_FILENAME
+  check_sha256 "$QRENCODE_HASH" "$QRENCODE_FILENAME"
+  bsdtar --no-same-owner --no-same-permissions -xf "$QRENCODE_FILENAME"
   rm qrencode*.tar.bz2
```
Comment (nurupo) [inline windows/cross-compile/build.sh]: Use `_FILENAME` here.

**Comment by nurupo** [comment]
:lgtm_strong:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4980#-:-L6xb1xpRefi35_qjZWH:b-jbsf8z)*

**Pull request merged**